cmd/subnetcmd/upgradecmd/install.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package upgradecmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ava-labs/avalanche-cli/pkg/binutils"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/subnet/upgrades"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	ANRclient "github.com/ava-labs/avalanche-network-runner/client"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+const (
+	timestampFormat  = "20060102150405"
+	tmpSnapshotInfix = "-tmp-"
+)
+
+// avalanche subnet upgrade generate
+func newUpgradeInstallCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install [subnetName]",
+		Short: "Installs upgrade bytes onto subnet nodes",
+		Long:  `Installs upgrade bytes onto VMs.`, // TODO fix this wording
+		RunE:  installCmd,
+		Args:  cobra.ExactArgs(1),
+	}
+	return cmd
+}
+
+func installCmd(cmd *cobra.Command, args []string) error {
+	subnetName := args[0]
+
+	if !app.SubnetConfigExists(subnetName) {
+		return errors.New("subnet does not exist")
+	}
+
+	sc, err := app.LoadSidecar(subnetName)
+	if err != nil {
+		return fmt.Errorf("unable to load sidecar: %w", err)
+	}
+
+	networkToUpgrade, err := selectNetworkToUpgrade(sc)
+	if err != nil {
+		return err
+	}
+
+	switch networkToUpgrade {
+	// in this case, we just are going to generate new update bytes
+	case futureDeployment:
+		return upgradeGenerateCmd(cmd, args)
```

Comment:
Instead of generating new upgrade bytes, should we suggest the user changes their precompile configuration and redeploys from scratch? Say it's much safer than using upgrade bytes

Reply:
Maybe this should just edit the precompile config in the genesis?

Reply:
I say if we are not going to allow to generate new bytes, then we should not allow this option in the first place. The use case becomes confused (edit in the genesis? redeploy from scratch - but maybe they actually never deployed in the first place, so what does it mean? etc.)

Reply:
To clarify my position: This is the `apply` command - it will just apply whatever json it finds on the network.

It is not the task of the `apply` command to identify the state of the `upgrade.json` file. 

`futureDeployment` to me clearly signals something which doesn't exist yet and is going to be done in the future. If we want to even allow this option _for the `apply` command_ , it's only for non-existing updates scenario IMO. Alternatively we disable `futureDeployment` for `apply` and that's it.

Reply:
Yeah, just remove this option

Reply:
Removed the option but had to refactor `selectNetworkToUpgrade` a bit so it would take a default option list and send an empty one here.